### ISSUE TITLE
No longer able to run onnxruntime_training_mnist sample

### ISSUE THREAD

**Issue by georgen117** [microsoft/onnxruntime#5839]
**Describe the bug**
I followed instructions posted by @liqunfu in #3706 to successfully run the provided mnist training sample.

> I attached the mnist test data and models. Please run with provided command line. As you can see, to work with cpp API, you need to follow the mnist example to build the graph with loss output so that backprop graph can be constructed.
>
> mnist.zip
> ./onnxruntime_training_mnist --model_name ~/mnist/mnist_gemm_simple --train_data_dir ~/mnist/mnist_data/

The sample will not run failing 
with the following error:

    Fail: Could not find an implementation for the node global_gradient_norm:ReduceAllL2(1)

Same error seen on Linux and Windows.

I have been able to run this sample successfully until recently.  Using git bisect I was able to identify this commit as the source of the failure:

    commit 07dc25e9390cf37c043c862de3ab41d266cca695
    Author: Sherlock <baihan.huang@gmail.com>
    Date:   Wed Nov 11 21:10:34 2020 -0800
        Compute global gradient norm according to 'enable_grad_norm_clip' (#5728)
        * Introduce PassThrough op to wait for all gradient ready before weight update
        * Compute gradient norm for fp32 runs
        * Update FE UT expected value
        * Respect enable_grad_norm_clip

**System information**
- seen on Windows 10 and Linux Ubuntu 18.04.5 LTS
- ONNX Runtime build from source
- ONNX Runtime version: 07dc25e9390cf37c043c862de3ab41d266cca695
- Python version: NA
- Visual Studio version (if applicable):  Visual Studio 2019 Version 16.7.7
- GCC/Compiler version (if compiling from source): gcc version 7.5.0
- CUDA/cuDNN version: NA
- GPU model and memory: NA

**To Reproduce**
Checkout commit 07dc25e9390cf37c043c862de3ab41d266cca695 or later

build code with training:

Follow instructions posted by @liqunfu in #3706 to download simple mnist model and training data.
for this example data placed in C:\mnist directory.

Windows:

    .\build.bat --config RelWithDebInfo --parallel  --build_shared_lib --cmake_generator "Visual Studio 16 2019" --enable_training
    cd build\Windows\RelWithDebInfo\RelWithDebInfo
    .\onnxruntime_training_mnist.exe --model_name C:\mnist\mnist_gemm --train_data_dir C:\mnist\mnist_data\

Linux:

    ./build.sh --config RelWithDebInfo --parallel  --build_shared_lib --enable_training
    cd build/Llnux/RelWithDebInfo
    export LD_Library_path=<pwd>
     ./onnxruntime_training_mnist --model_name ~/mnist/mnist_gemm_simple --train_data_dir ~/mnist/mnist_data/

**Expected behavior**
Expect the onnxruntime_training_mnist sample to run to completion with a precision of about 0.89

**Additional context**
all context should be represented above

**Comment by mrry** [comment]
Thanks for the detailed report, and sorry for the breakage!

I think the recently merged pull request #5833 should fix the problem (thanks @iK1D!), so you should be able to update to the latest master and train MNIST again.

**Comment by gnash117** [comment]
Thanks, just updated with master and I am able to run MNIST again. :)

**Comment by georgen117** [comment]
Thanks @mrry I am able to run the MNIST sample again.
